docker/importer/importer_test.py

```diff
@@ -554,6 +560,92 @@ def test_import_override(self, unused_mock_time: mock.MagicMock,
             for x in upload_from_str.call_args_list))
 
 
+@mock.patch('importer.utcnow', lambda: datetime.datetime(2024, 1, 1))
+class RESTImporterTest(unittest.TestCase):
+  """REST importer tests."""
+  httpd = None
+
+  def setUp(self):
+    tests.reset_emulator()
+    self.maxDiff = None  # pylint: disable=invalid-name
+    self.tmp_dir = tempfile.mkdtemp()
+
+    tests.mock_datetime(self)
+    self.mock_repo = tests.mock_repository(self)
+    warnings.filterwarnings("ignore", "unclosed", ResourceWarning)
+
+    storage_patcher = mock.patch('google.cloud.storage.Client')
+    self.addCleanup(storage_patcher.stop)
+    self.mock_storage_client = storage_patcher.start()
+
+    self.remote_source_repo_path = self.mock_repo.path
+    self.source_repo = osv.SourceRepository(
+        type=osv.SourceRepositoryType.REST_ENDPOINT,
+        id='curl',
+        name='curl',
+        link=f'http://{SERVER_ADDRESS[0]}:{SERVER_ADDRESS[1]}',
+        rest_api_url=f'http://{SERVER_ADDRESS[0]}:{SERVER_ADDRESS[1]}/',
+        db_prefix='CURL-',
+        extension='.json',
+        editable=False)
+    self.source_repo.put()
+    self.tasks_topic = f'projects/{tests.TEST_PROJECT_ID}/topics/tasks'
+
+  def tearDown(self):
+    shutil.rmtree(self.tmp_dir, ignore_errors=True)
+    self.httpd.shutdown()
+
+  @mock.patch('google.cloud.pubsub_v1.PublisherClient.publish')
+  @mock.patch('time.time', return_value=12345.0)
+  def test_basic(self, unused_mock_time: mock.MagicMock,
+                 mock_publish: mock.MagicMock):
+    "Testing basic rest endpoint import"
+    self.httpd = http.server.HTTPServer(SERVER_ADDRESS, MockDataHandler)
+    thread = threading.Thread(target=self.httpd.serve_forever)
+    thread.start()
+    self.source_repo.last_update_date = datetime.datetime(2020, 1, 1)
+    self.source_repo.put()
+    imp = importer.Importer('fake_public_key', 'fake_private_key', self.tmp_dir,
+                            importer.DEFAULT_PUBLIC_LOGGING_BUCKET, 'bucket',
+                            False)
+    imp.run()
+    mock_publish.assert_called()
```

Comment:
Can you assert the arguments that are called here as well? i.e. assert_called_with

Reply:
Because there is ~24 vulns called here, I've now asserted the number called. This test expects them all to be updated so it now checks how many vulns are in the given file to compare and assert. For the filtered results (in the other testcase), they have all the mock calls asserted.